src/misc_gui.cpp

```diff
@@ -981,6 +981,23 @@ struct QueryStringWindow : public Window
 			resize->width = 0;
 			size->width = 0;
 		}
+
+		if (widget == WID_QS_WARNINGBOX && (this->flags & QSF_PASSWORD) == 0) {
+			/* We don't want this widget to show! */
+			fill->width = 0;
+			resize->width = 0;
+			size->width = 0;
+		}
+		if (widget == WID_QS_WARNING) {
+			*size = GetStringBoundingBox(STR_WARNING_PASSWORD_SECURITY);
```

Comment:
Were you looking for `GetStringMultiLineBoundingBox`? It will always keep the width and only adjust the height, which should mean no manual line breaks.

Reply:
I tried that. Problem is that the width of the rest of the window is unknown at this point, so I don't have a reference width to break lines against.

Reply:
That's basically a problem anywhere we use that function :/